### PR TITLE
Low level configuration functions for S3

### PR DESCRIPTION
**Pull request by amercader**
@roll 1st PR for S3 stuff. I think it makes sense to review/merge bits at a time rather than the whole lot but let me know if you want to review everything in one go.

This includes two clients that use boto3 to interact with the S3 Buckets / Lambdafunction as described in https://github.com/frictionlessdata/goodtables.io/issues/58#issuecomment-272159815

Configuration needed from env vars.

Lots of corner cases, extensive tests

### CODE REVIEW

File: goodtablesio/plugins/s3/utils.py [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+            secret_access_key (str): The Secret Access Key part of the
+                credentials
+
+        """
+        self.client = boto3.client(
+            's3',
+            aws_access_key_id=access_key_id,
+            aws_secret_access_key=secret_access_key)
+
+    def _notification_id_for_bucket(self, bucket_name):
```
Comment (roll) [inline goodtablesio/plugins/s3/utils.py]: It's style related but I suppose it's easier to understand class API if public methods goes first then private methods.
